### PR TITLE
Add FLUKA and FLUKA_VMC

### PR DESCRIPTION
**Pull request by dberzano**
* A fix to the ROOT recipe was needed to specify the Fortran compiler

### CODE REVIEW

**Comment by dberzano** [comment]
This has been successfully tested with ROOT 5 and ROOT 6.

**Comment by dberzano** [comment]
@chiarazampolli @pzhristov

**Comment by pzhristov** [comment]
Hi @dberzano 
We have to provide in the installation all the files with extension .bin They contain the Fluka data for all the processes.

Cheers,
Peter

**Comment by dberzano** [comment]
Thanks, I’ll fix the recipe.
-- 
--
Dario Berzano
about.me/dario.berzano

**Comment by dberzano** [comment]
FLUKA is not free software and the ALICE collaboration has been granted a special license for using it.

* The repository containing FLUKA code referenced in this recipe is protected and may be accessed only by specific users.
* Using FLUKA with the VMC is possible due to an agreement between the FLUKA authors and the ALICE collaboration.
* Deployment of this software for internal ALICE tests is compliant with the agreement between the FLUKA authors and the ALICE collaboration.

In other words, the sole possession of these software _recipes_ does not enable you by any means to install FLUKA freely. FLUKA installation is subject to restrictions, which we enforce.

The points above were clarified with the ALICE contact persons with the FLUKA authors: @pzhristov and @amorsch.

**Comment by dberzano** [comment]
Hi all, turns out we also need some `.dat` files. The recipe has been updated.

**Comment by chiarazampolli** [comment]
Hello Dario,

Did you also include the export of the Fluka directory in $ALICE_ROOT?

Cheers,

Chiara

**Comment by dberzano** [comment]
Hi @chiarazampolli, I see you write:

> Add TFluka to $ALICE_ROOT, so that then I can do:
> `ln -fs $ALICE_ROOT/TFluka/input/coreFlukaVmc.inp coreFlukaVmc.inp`

but I did not understand what should I do and where.

**Comment by dberzano** [comment]
Let me specify: `TFluka` is already in AliRoot, and the aforementioned file is available. You can do that `ln` command already, provided you load both FLUKA_VMC and AliPhysics as packages.

**Comment by chiarazampolli** [comment]
Ciao Dario,

Actually the file I need (coreFlukaVmc.inp) is both in AliRoot/TFluka/input (see https://github.com/alisw/AliRoot/tree/master/TFluka/input) and in Fluka itself:

ls /cvmfs/alice-nightlies.cern.ch/x86_64-2.6-gnu-4.1.2/Packages/FLUKA_VMC/2011.2c-vmc1-2/input/
coreFlukaVmc.inp

I don't know which of the two files is correct, if they are identical, etc. So I would remove one of the two. If we remove the one in Fluka, we need to have a way to get the one from AliRoot, which should be something like:

ln -fs $ALICE_ROOT/TFluka/input/coreFlukaVmc.inp coreFlukaVmc.inp

but $ALICE_ROOT does not contain TFluka:

ls $ALICE_ROOT
ACORDE	  bin	 EMCAL	EVGEN	 FMD  HMPID    ITSMFT  libexec	PARfiles  PYTHIA6  relocate-me.sh  STEER  TEvtGen  TRD		 TUHKMgen
AD	  cmake  etc	FASTSIM  GRP  include  LHAPDF  macros	PHOS	  PYTHIA8  STARLIGHT	   T0	  TOF	   TRIGGER	 VZERO
ANALYSIS  data	 EVE	FIT	 HLT  ITS      lib     MUON	PMD	  QAref    STAT		   test   TPC	   TTherminator  ZDC

Cheers,

Chiara

**Comment by dberzano** [comment]
The two files are different. But they differ for a little bit only:

```diff
--- AliRoot/TFluka/input/coreFlukaVmc.inp	2018-01-16 10:49:40.061725224 +0100
+++ FLUKA_VMC/input/coreFlukaVmc.inp	2018-03-22 23:26:44.377143586 +0100
@@ -1,5 +1,7 @@
-GLOBAL        10000.                           4.0
+GLOBAL        10000.
 DEFAULTS                                                              NEW-DEFA
+*OPEN             15.                                                  UNKNOWN
+*FlukaVmc.err
 OPEN              1.                                                  OLD
 random.dat
 GEOBEGIN                                                              COMBINAT
@@ -7,7 +9,6 @@ GEOEND
 *
 *...+....1....+....2....+....3....+....4....+....5....+....6....+....7....+
 *
-EVENTYPE                             1.0
 BEAM          7000.0                                                  MUON-
 MGNFIELD        10.0
 SOURCE
```

I haven't got the faintest idea about what those parameters do, but I'd keep the one that comes with `FLUKA_VMC` as it's newer, and do nothing on the AliRoot side (which requires opening a PR to AliRoot, not alidist).

Can someone help here understanding if we can safely retain and use `coreFlukaVmc.inp` from FLUKA_VMC?

**Comment by amorsch** [comment]
Please use coreFlukaVmc.inp	from AliRoot/TFluka
It contains the changes necessary to run with the latest version.
Cheers Andreas

**Comment by dberzano** [comment]
So, with the latest change I have committed here, the file is accessible from `$FLUVMC/input/coreFlukaVmc.inp`.

**Comment by chiarazampolli** [comment]
Ciao Dario,
Thanks! There is also a variable called FLUPRO that one can use, correct?
Cheers,
Chiara

**Comment by dberzano** [comment]
For the record I am building a version (on nightlies) that you can use.

**Comment by chiarazampolli** [comment]
Ok, thanks!
Chiara

**Comment by sawenzel** [comment]
+1 for new CI.

File: fluka.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  - "GCC-Toolchain:(?!osx)"
+env:
+  FLUPRO: "$FLUKA_ROOT"
+  FLUFOR: "gfortran"
+prepend_path:
+  PATH: "$FLUKA_ROOT:$FLUKA_ROOT/flutil"
+---
+#!/bin/bash -e
+
+if [[ $(uname) != Linux || $(uname -m) != x86_64 ]]; then
```
Comment (ktf) [inline fluka.sh]: Don't use uname, use $ARCHITECTURE or whatever variable we used to set.

**Comment by amorsch** [comment]
+1

File: fluka.sh (hunk 1 above)
Comment (dberzano) [reply fluka.sh]: Thanks for spotting it, I have fixed it.
